config/initializers/timeout.rb

```diff
@@ -0,0 +1,14 @@
+require 'net/http'
+
+# Lengthen timeout in Net::HTTP
+# Todo: not needed in Avalon v7.1
+module Net
+    class HTTP
+    alias old_initialize initialize
+
+        def initialize(*args)
+            old_initialize(*args)
+            @read_timeout = ENV['AVALON_NET_HTTP_READ_TIMEOUT'].to_i if ENV['AVALON_NET_HTTP_READ_TIMEOUT']
+        end
+    end
+end
```

Comment:
Do you know whether this is being called in the upload path in Avalon?

[Because this can be set per-instance](https://ruby-doc.org/stdlib-2.6.3/libdoc/net/http/rdoc/Net/HTTP.html#method-i-read_timeout-3D), I'm wondering if it wouldn't be better to monkey-path this in the Avalon caller instead? Just on the basis of it is a more limited change, impacts fewer things, etc. As it stands if someone were to crank this timeout really high to accommodate a huge file ("someone wants to upload a 500 gig file? let's just crank this env variable up to 5000"), it has the side-effect of impacting every other use of Net::HTTP as well, which could impact things like unrelated calls to external APIs, or how long a browser with a slow connection can occupy a connection before timing out (which, in turn, increase the ease with which you can DOS the site, and decreases the scalability of the site, etc).

If we can't figure it out, then this is probably the best we can do, but if there's a more tightly-scoped monkey-patch target I think we should try to go for that instead?

Reply:
I'll try targeting the monkey patch to the rubyhorn gem (the gem implementing the rest call to Matterhorn/transcoding service). 
Possibly here: 
https://github.com/avalonmediasystem/rubyhorn/blob/master/lib/rubyhorn/matterhorn_client.rb
or somewhere else in the rubyhorn gem.

I like the above narrowing idea to decrease the chance of unintended negative side-effects of the broader approach.

Reply:
Ah, ok. It looks like Matterhorn uses this (somewhat out of date) wrapper gem: https://github.com/rest-client/rest-client

`RestClient::Request` takes a `:timeout` option that sets the read timeout in `Net::HTTP`: https://github.com/rest-client/rest-client/blob/master/lib/restclient/request.rb#L93

`RestClient::Resource` wraps request and passes its option block down to request, here: https://github.com/rest-client/rest-client/blob/master/lib/restclient/resource.rb#L39

So I _think_ you could patch it right here: https://github.com/avalonmediasystem/rubyhorn/blob/master/lib/rubyhorn/rest_client/common.rb#L9 by passing in a `:timeout` option to the initializer

Reply:
On the same page. Did a quick test, appears feasible. Trying the `:open_timeout` as it's a bit narrower.

Reply:
@mbarnett ready for a review when you have a chance. Monkey patching is not one of my Ruby skills.